zone/embperl.cpp

```diff
@@ -273,8 +273,24 @@ int Embperl::dosub(const char * subname, const std::vector<std::string> * args,
 	FREETMPS;
 	LEAVE;
 
-	if(error.length() > 0)
-	{
+	// not sure why we pass this as blind args, strange
+	// check for syntax errors
+	if (args && !args->empty()) {
+		const std::string &filename = args->back();
+		std::string sub = subname;
+		if (sub == "main::eval_file" && !filename.empty() && File::Exists(filename)) {
+			std::string syntax_error = Process::execute(
+				fmt::format("perl -c {} 2>&1", filename)
```

Comment:
If we can easily leverage the API in a way that you know of offhand, lets do it. Otherwise this doesn't seem to be too expensive of an operation.

I'll attach measurements in regular comments.

Really good call out though

Reply:
Add
```cpp
"	die $@ if ($@ && $@ !~ /did not return a true value/);"
```
To the function in init_eval_file after the eval line at https://github.com/EQEmu/Server/blob/ae4908b40c12d204ba633c619cfc5eb6fc732967/zone/embperl.cpp#L213

This should fix errors not showing up without needing to invoke the external process. I assume this was disabled to avoid needing all scripts to have `1;` at the end since that function loads script via require. The error string exclusion looks like what newer versions of perl might be experimenting with as an option to remove that requirement (`module_true`).

In the future we could also replace these injected perl functions and just use perlbind to load the script files which would catch errors. I'd rather not do that now since that feature had minimal testing and I don't want to have to look at in case of regressions.